### PR TITLE
finish director_privileges impl + test

### PR DESCRIPTION
**Pull request by amihaiemil**
PR for #284 

* finished director_privileges check according to the puzzle
* added test
* **note** that the Employee <-> Role relation had already been added in this commit: 62d9c69f938005be4d3903d1a88562e734d156a0

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#427](https://github.com/timelesslounge/timelessis/issues/427) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#427](https://github.com/timelesslounge/timelessis/issues/427) is assigned to @vryazanov[/z](https://www.0crat.com/u/vryazanov), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/4b96eea9-4518-4a78-9b94-1bb5fb78b52d); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: tests/integration/it_director_privilages_test.py [hunk 1]
```diff
@@ -71,3 +86,89 @@ def test_cant_access_other_company_employees(app, db_session):
         method=Method.READ, resource="employee", employee_id=other.id
     )
 
+
+def test_can_access_subalterns(app, db_session):
```
Comment (vryazanov) [inline tests/integration/it_director_privilages_test.py]: @amihaiemil I suggest to use factories to make the code simpler and shorter, isee how it's done in https://github.com/timelesslounge/timelessis/blob/master/tests/integration/it_manager_privilages_test.py#L94

File: timeless/access_control/director_privileges.py [hunk 2]
```diff
@@ -12,29 +11,35 @@ def has_privilege(method=None, resource=None, *args, **kwargs) -> bool:
...
+    profile of the employees with a lower role, who work at the same company.
+    """
+    if employee_id == user.id:
         return True
     else:
         other = Employee.query.get(employee_id)
-        return user.company_id == other.company_id
+        other_role = other.role.name
+        return user.company_id == other.company_id and\
+            user.role.name == "director" and\
```
Comment (vryazanov) [inline timeless/access_control/director_privileges.py]: @amihaiemil I think should hardcode all names directly in `Role` model (see https://github.com/timelesslounge/timelessis/blob/master/timeless/roles/models.py#L22). I left a puzzle there to find a nice way to handle role types, and I think while it's not resolved we should stay all hardcoded names there in the model. in order to make it easier to change in the future without searching any other places where it's used. wdyt?

**Comment by amihaiemil** [comment]
@vryazanov I see, thanks for the comments. I'll fix them today, or tomorrow morning the latest. Sorry for the delay!

**Comment by amihaiemil** [comment]
@vryazanov fixed, pls have a look :)

**Comment by vryazanov** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @vryazanov[/z](https://www.0crat.com/u/vryazanov), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Job [#427](https://github.com/timelesslounge/timelessis/issues/427) is not in the agenda of @vryazanov[/z](https://www.0crat.com/u/vryazanov), can't inspect

**Comment by 0crat** [comment]
The job [#427](https://github.com/timelesslounge/timelessis/issues/427) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
@vryazanov According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only two issues were found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by vryazanov** [comment]
@ypshenychka  I confirm

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @vryazanov[/z](https://www.0crat.com/u/vryazanov)

**Comment by 0crat** [comment]
Quality review completed: +6 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**